### ISSUE TITLE
[Feature Request] Folders bar dark mode toggle

### ISSUE THREAD

**Issue by Sasha-Sorokin** [telegramdesktop/tdesktop#7460]
### Describe the solution you'd like

Back in the days when GitHub first introduced its header redesign, they faced some backlash by the designers and users, telling that this is not a very accessible solution. Number of extensions and user styles arose, and people like me still dislike it by this day, but that doesn't bother me as much because I mostly not at the top of the page. Telegram on the opposite has this sidebar always visible, so it bothers me and the others highly probably.

A popular messaging platform among the gamers and some other communities, Discord, also had dark sidebar for a long time in light theme and it was really hated by the users as seen by enormous number of feedback site ideas and tweets. When they came up with the redesign of light theme, they faced yet another wave of complaints, this time it was for its removal — some users found it's more accessible, some just liked that. The solution from the Discord team was brilliant: they simply added a toggle in settings which toggles dark sidebar on and off. Everyone's happy and the life goes on… maybe. The redesign still wasn't well received, but that's another story.

I propose that we actually learn from these experiences and make the same changes as Discord: for the standard light themes there should be a toggle which makes sidebar dark, right next to accents. For the standard dark themes light sidebar will probably look bad, so toggle is unnecessary and maybe should collapse (just as accents when you set to classic theme). And the custom themes just should have all these colors changeable as they see fit.

*Did I mentioned sidebar completely ignores accent colors?*

### Describe alternatives you've considered

- Closing my eyes. Doesn't help and even produces worse results — can't see anything anymore. Opened them back.

- Not using folders. Very inefficient, wouldn't recommend.

- Talking to therapist. Said it's only a beta and things may change, but I, very concerned about it, should really write it down and send to developers. So there it is.

### Additional context

**This feedback is for the folders feature that is currently in beta**.

**Comment by Brawaru** [comment]
While you now can customize sidebar colours via custom theme, a simpler toggle for standard themes would be very much appreciated.

**Comment by Brawaru** [comment]
Stalen't.

**Comment by kass-kass** [comment]
+10000000 on this, I literally don't use folders because of how horrible they look on the desktop client. If the bar wasn't dark (not a fan of dark UI in general), or was stylable in any way, or even just possible to hide, it would make things way better. I'm sure I'm not the only one missing out on a great major feature because of bad UI choices.

**Comment by Teklars** [comment]
I just started using Folders, yeah so strange it's still not fixed, Folders sidebar on desktop client is dark and on light theme it looks annoying and more like a bug in software, in night mode it's ok,  please fix it so in day mode it would be light

**Comment by Aokromes** [comment]
> I just started using Folders, yeah so strange it's still not fixed, Folders sidebar on desktop client is dark and on light theme it looks annoying and more like a bug in software, in night mode it's ok, please fix it so in day mode it would be light

workaround: use a theme with light folder bar

**Comment by Teklars** [comment]
> > I just started using Folders, yeah so strange it's still not fixed, Folders sidebar on desktop client is dark and on light theme it looks annoying and more like a bug in software, in night mode it's ok, please fix it so in day mode it would be light
> 
> workaround: use a theme with light folder bar

so far I see themes doesn't affect folder bar, it stays dark

**Comment by ctp9** [comment]
Here to confirm that this is still an issue - folders bar is stuck on dark mode and does not adjust when switched to day mode.

**Comment by ctp9** [comment]
Here to confirm that this is still an issue - folders bar is stuck on night mode and does not change when switched to day mode.

**Comment by ctp9** [comment]
Guys, it's been over a year. Change the color of the folders bar to match operating system theme. Is it really that difficult? Maybe I'm just ignorant.

**Comment by ctp9** [comment]
Guys, it's been over a year. Change the color of the folders bar to match operating system theme / Telegram theme. Is it really that difficult? Maybe I'm just ignorant.

**Comment by ilya-fedin** [comment]
@ctp9 the fact a feature request exists means only that someone wants such a feature, it doesn't mean such a feature will be considered to be implemented, ever.

**Comment by ctp9** [comment]
> @ctp9 the fact a feature request exists means only that someone wants such a feature, it doesn't mean such a feature will be considered to be implemented, ever.

Okay, now that you say this - I think we need to escalate this "feature request" to an official "bug"... because this isn't a "feature" that we should be "requesting"...

This is something that needs to be working right out of the gate when one switches between Day/Night themes - the folders bar must switch colors accordingly. It is a considered a bug if the colors of the folders bar do not follow the theme setting.

**Comment by ctp9** [comment]
> @ctp9 the fact a feature request exists means only that someone wants such a feature, it doesn't mean such a feature will be considered to be implemented, ever.

Okay, now that you say this - I think we need to escalate this "feature request" to an official "bug"... because this isn't a "feature" that we should be "requesting"...

This is something that needs to be working right out of the gate when one switches between Day/Night themes - the folders bar must switch colors accordingly. It should be considered a UI bug if the colors of the folders bar do not follow the theme setting.

**Comment by ilya-fedin** [comment]
This won't be accepted as a bug report anyway since this is intended

**Comment by ctp9** [comment]
> This won't be accepted as a bug report anyway since this is intended

So you say it is "intended". Please provide supporting evidence of the intention for the folders bar remaining dark and not adjusting with Day/system theme. That way, once we know the intention that you speak of, we can lay this idea of the folders bar adjusting colors dynamically to rest, and accept it as a self-evident feature of Telegram...

![Telegram Theme Bug](https://user-images.githubusercontent.com/15347630/146279358-244d245d-f732-4a32-91e7-d496d0bd8e13.png)

**Comment by ilya-fedin** [comment]
> Please provide supporting evidence of the intention for the folders bar remaining dark and not adjusting with Day/system theme.

I don't have such information. I only know that it's intended.

**Comment by ctp9** [comment]
Yeah... not very convincing. I still think it needs to be revised and considered a UI bug. Time to evolve with the times and start making user interfaces consistent with the themes of the system and the application itself.

I'm sure I'm not the only one who thinks this folders bar is a major eyesore and is completely out of place looking dark on a Day theme / Light system theme.

**Comment by ctp9** [comment]
It looks like sh*t. Don't continue telling Telegram's userbase that this is by design. Time to re-evaluate and adapt to modern UI expectations. We aren't stupid - our eyes are telling us that this is bad design. We need to get this considered a UI bug at this point.

**Comment by ctp9** [comment]
So... what's happening here? Meanwhile every single competitor app is properly utilizing material/fluid design elements - Telegram Desktop is stuck in the dark ages. Let's please get this fixed. All it takes is a day's worth of work to get this done.

**Comment by 23rd** [comment]
If this is such a significant problem for you, why don't you change the color of the folder bar for yourself?
If it is all about colors, why it can't be done with custom themes?

**Comment by ctp9** [comment]
> If this is such a significant problem for you, why don't you change the color of the folder bar for yourself? If it is all about colors, why it can't be done with custom themes?

You know what - you are right. I could have just edited the theme... which I didn't think was possible on the desktop version until you mentioned it. With that said - all it took me was <1 minute of work to change the sidebar color to match the Day theme.

What's so difficult for the devs to do this too? Honestly tell please me - why is everyone here resisting making this change permanent in the Day theme? I am asking this with good intentions... not trying to stir up any conflict.

![Telegram Theme Bug Manually Fixed](https://user-images.githubusercontent.com/15347630/174194236-3a6ab27d-eda0-4c1c-aebe-1ad2c5d6ba9d.png)

**Comment by Aokromes** [comment]
> > If this is such a significant problem for you, why don't you change the color of the folder bar for yourself? If it is all about colors, why it can't be done with custom themes?
> 
> You know what - you are right. I could have just edited the theme... which I didn't think was possible on the desktop version until you mentioned it. With that said - all it took me was <1 minute of work to change the sidebar color to match the Day theme.

i said that 1 year ago...

**Comment by ilya-fedin** [comment]
> why is everyone here resisting making this change permanent in the Day theme?

Default themes look like Telegram designers see them

**Comment by john-preston** [comment]
@ctp9 Because the initial design mockups provided to me looked exactly the way the app looks. So "devs" didn't do this yet because there is no intention to do that - the default colors are the ones that are supposed to be.

**Comment by Brawaru** [comment]
Kinda sorry for this guys, I didn't intend for this issue to house such entitlement, I just wanted to ask for a feature...

**Comment by ctp9** [comment]
> Because the initial design mockups provided to me looked exactly the way the app looks. So "devs" didn't do this yet because there is no intention to do that - the default colors are the ones that are supposed to be.

Well... thank you for responding. I still don't understand - none of these reasons and justifications make any "sense" to me. But nonetheless I appreciate you finally explaining your position. I am glad there is at least the option to manually edit themes.

> Kinda sorry for this guys, I didn't intend for this issue to house such entitlement, I just wanted to ask for a feature...

Relax buddy. Just having a conversation here and trying to get some answers from the devs - there's no entitlement. The conversation is had with good intentions and in good faith... some of us are suggesting improvements for the app that can benefit everyone else who is using it.

**Comment by Aokromes** [comment]
> > > I just started using Folders, yeah so strange it's still not fixed, Folders sidebar on desktop client is dark and on light theme it looks annoying and more like a bug in software, in night mode it's ok, please fix it so in day mode it would be light
> > 
> > 
> > workaround: use a theme with light folder bar
> 
> so far I see themes doesn't affect folder bar, it stays dark

https://t.me/addtheme/kd14jziGvcRuiuIC a theme with folder bar white.
